kv/README.md

```diff
@@ -0,0 +1,67 @@
+# Proposed design for JetStream based KV services
+
+This is a proposal to design KV services ontop of JetStream, this requires a KV client to be written, the basic
+features are:
+
+ * Multiple named buckets full of keys with n historical values kept per value
+ * Put and Get `string(k)=string(v)` values
+ * Deleting a key completely
+ * Per key TTL
+ * Compacting a key down to n remaining values
+ * Watching a specific key or the entire bucket for live updates
+ * Encoders and Decoders that transforms both keys and values
+ * A read-cache that builds an in-memory cache for fast reads
+ * Ability to read from regional read replicas maintained by JS Mirroring (planned)
+ * read-after-write safety unless the read cache of read replicas were used
+ * Valid keys are `^[a-zA-Z0-9_-/]+$` after encoding (not validated now)
+ * Valid buckets are `^[a-zA-Z0-9_-]+$` (not validated now)
+
+This is an implementation of this concept focussed on the CLI, nats.go and others will have to build language
+specific interfaces focussed on performance and end user.
+
+## Design
+
+### Storage
+
+Given a bucket `CONFIGURATION` we will have:
+
+ * A stream called `KV_CONFIGURATION` with subjects `kv.CONFIGURATION.*`
```

Comment:
Need to decide on this prefix in stream name and in subjects, I dont really want it configurable but probably we'd have to for support of imports and stuff

Reply:
made configurable